services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaServiceITest.kt

```diff
@@ -1572,8 +1570,92 @@ class HankeKayttajaServiceITest : DatabaseTest() {
+                prop(HankekayttajaEntity::puhelin).isEqualTo(update.puhelinnumero)
+            }
+        }
+    }
+
+    @Nested
+    inner class UpdateContactInfo {
+        private val update = ContactUpdate("updated@email.test", "9991111")
+
+        @Test
+        fun `Throws exception if there's no hanke`() {
+            val hanketunnus = "HAI-001"
+
+            val failure = assertFailure {
+                hankeKayttajaService.updateContactInfo(hanketunnus, update, UUID.randomUUID())
+            }
+
+            failure.all {
+                hasClass(HankeNotFoundException::class)
+                messageContains(hanketunnus)
+            }
+        }
+
+        @Test
+        fun `Throws exception if user not in hanke`() {
+            val hanke = hankeFactory.builder(USERNAME).save()
+            val otherHanke = hankeFactory.builder("Other user").save()
+            val otherUser = kayttajaFactory.saveIdentifiedUser(otherHanke.id)
+
+            val failure = assertFailure {
+                hankeKayttajaService.updateContactInfo(hanke.hankeTunnus, update, otherUser.id)
+            }
+
+            failure.all {
+                hasClass(HankeKayttajaNotFoundException::class)
+                messageContains(otherUser.id.toString())
+            }
+        }
+
+        @Test
+        fun `Throws exception when user is identified and try to change name`() {
+            val update = ContactUpdate("updated@email.test", "9991111", "Uusi", "Nimi")
+            val hanke = hankeFactory.builder(USERNAME).save()
+            val identifiedUser = kayttajaFactory.saveIdentifiedUser(hanke.id)
+
+            val failure = assertFailure {
+                hankeKayttajaService.updateContactInfo(hanke.hankeTunnus, update, identifiedUser.id)
+            }
+
+            failure.all {
+                hasClass(UserAlreadyHasPermissionException::class)
+                messageContains("The user already has an active permission")
+            }
+        }
+
+        @Test
+        fun `Updates email and phone number of a identified user`() {
+            val hanke = hankeFactory.builder(USERNAME).save()
+            val identifiedUser = kayttajaFactory.saveIdentifiedUser(hanke.id)
+
+            hankeKayttajaService.updateContactInfo(hanke.hankeTunnus, update, identifiedUser.id)
+
+            val kayttaja = hankeKayttajaService.getKayttajaForHanke(identifiedUser.id, hanke.id)
+            assertThat(kayttaja).isNotNull().all {
+                prop(HankekayttajaEntity::etunimi).isEqualTo(identifiedUser.etunimi)
+                prop(HankekayttajaEntity::sukunimi).isEqualTo(identifiedUser.sukunimi)
+                prop(HankekayttajaEntity::sahkoposti).isEqualTo(update.sahkoposti)
+                prop(HankekayttajaEntity::puhelin).isEqualTo(update.puhelinnumero)
+            }
+        }
+
+        @Test
+        fun `Updates email, phone and name of an unidentified user`() {
```

Comment:
There could also be a test for updating just the email and phone number of an unidentified user.